homeassistant/components/homekit/type_sensors.py

```diff
@@ -128,21 +153,244 @@ def __init__(self, *args):
         """Initialize a AirQualitySensor accessory object."""
         super().__init__(*args, category=CATEGORY_SENSOR)
 
-        serv_air_quality = self.add_preload_service(
-            SERV_AIR_QUALITY_SENSOR, [CHAR_AIR_PARTICULATE_DENSITY]
+        """determine the charciteristics of the sensor based on the curent attributes"""
+        chars = []
+        current_state = self.hass.states.get(self.entity_id)
+        nitrogen_dioxide_density = convert_to_float(
+            current_state.attributes.get(ATTR_NITROGEN_DIOXIDE_DENSITY)
+        )
+        voc_density = convert_to_float(current_state.attributes.get(ATTR_VOC_DENSITY))
+        pm_2_5_density = convert_to_float(
+            current_state.attributes.get(ATTR_PM_2_5_DENSITY)
         )
-        self.char_quality = serv_air_quality.configure_char(CHAR_AIR_QUALITY, value=0)
-        self.char_density = serv_air_quality.configure_char(
-            CHAR_AIR_PARTICULATE_DENSITY, value=0
+        pm_10_density = convert_to_float(
+            current_state.attributes.get(ATTR_PM_10_DENSITY)
+        )
+        pm_density = convert_to_float(current_state.attributes.get(ATTR_PM_DENSITY))
+        pm_size = convert_to_float(current_state.attributes.get(ATTR_PM_SIZE))
+
+        if nitrogen_dioxide_density is not None:
+            chars.append(CHAR_NITROGEN_DIOXIDE_DENSITY)
+
+        if voc_density is not None:
+            chars.append(CHAR_VOC_DENSITY)
+
+        """
+        NOTE: this would be the easy way but it seems the driver is missing the PM2_5_Density key
```

Comment:
Does `PM2.5Density` work? (https://github.com/ikalchev/HAP-python/blob/cc716fa1443dcc0fae47368421a1d3543b13f9bb/pyhap/resources/characteristics.json#L869)

Reply:
As i was falling asleep last night i realized i had the name wrong. that worked pushed fix.

Reply:
fixed this.